core/block.go

```diff
@@ -34,3 +39,119 @@ func (b *Block) Hash() *felt.Felt {
 	// https://docs.starknet.io/documentation/develop/Blocks/header/#block_hash
 	return nil
 }
+
+type (
+	commitmentStorage     map[string]string
+	commitmentTrieStorage struct {
+		storage commitmentStorage
+	}
+)
+
+func (s *commitmentTrieStorage) Put(key *bitset.BitSet, value *TrieNode) error {
+	keyEnc, err := key.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	vEnc, err := value.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	s.storage[hex.EncodeToString(keyEnc)] = hex.EncodeToString(vEnc)
+	return nil
+}
+
+func (s *commitmentTrieStorage) Get(key *bitset.BitSet) (*TrieNode, error) {
+	keyEnc, _ := key.MarshalBinary()
+	value, found := s.storage[hex.EncodeToString(keyEnc)]
+	if !found {
+		panic("not found")
+	}
+
+	v := new(TrieNode)
+	decoded, _ := hex.DecodeString(value)
+	err := v.UnmarshalBinary(decoded)
+	return v, err
+}
+
+func (s *commitmentTrieStorage) Delete(key *bitset.BitSet) error {
+	keyEnc, _ := key.MarshalBinary()
+	delete(s.storage, hex.EncodeToString(keyEnc))
+	return nil
+}
+
+// TransactionCommitment is the root of a height 64 binary Merkle Patricia tree of the
+// transaction hashes and signatures in a block.
+func TransactionCommitment(transactions []*clients.Transaction) (*felt.Felt, error) {
```

Comment:
We should use core types here. If the core types are missing information, we can modify them.

Reply:
Or maybe we should move the commitment implementation to `clients`? What do you think?

Reply:
I think it would be best to have the commitment calculation performed on the core types. This way, when the feeder gateway types are deprecated, we don't need to change nearly as much code.

Reply:
Yeah, you're right. I didn't consider the depreciation of the feeder gateway types